x-pack/plugins/security_solution/public/timelines/components/timeline/tabs/esql/index.tsx

```diff
@@ -166,9 +180,14 @@ export const DiscoverTabContent: FC<DiscoverTabContentProps> = ({ timelineId })
       setDiscoverStateContainer(stateContainer);
       let savedSearchAppState;
       if (savedSearchId) {
-        const localSavedSearch = await savedSearchService.get(savedSearchId);
-        initializeLocalSavedSearch(localSavedSearch, timelineId);
-        savedSearchAppState = getAppStateFromSavedSearch(localSavedSearch);
+        try {
+          const localSavedSearch = await savedSearchService.get(savedSearchId);
+          initializeLocalSavedSearch(localSavedSearch, timelineId);
+          savedSearchAppState = getAppStateFromSavedSearch(localSavedSearch);
+        } catch (e) {
+          // eslint-disable-next-line no-console
+          console.error('Stale Saved search Id which no longer exists', e);
```

Comment:
Do we want to keep this log in prod?

Reply:
yep.. I was thinking the same and decided to keep it.